VolleyManagement.UI/Areas/WebAPI/Controllers/TournamentsController.cs

```diff
@@ -107,20 +107,14 @@ public List<PivotStandingsViewModel> GetTournamentPivotStandings(int id)
         [Route("api/Tournament/{tournamentId}/Schedule")]
         public ScheduleViewModel GetSchedule(int tournamentId)
         {
-            List<GameViewModel> gamesViewModel = _gameService.GetTournamentGames(tournamentId)
-                                                        .Select(t => GameViewModel.Map(t)).ToList();
-            foreach (var item in gamesViewModel)
-            {
-                if (item.Result.TotalScore.IsEmpty)
-                {
-                    item.Result = null;
-                }
-            }
+            var games = _gameService.GetTournamentGames(tournamentId)
+                                    .Select(GameViewModel.Map)
+                                    .ToList();
 
-            var resultGroupedByWeek = gamesViewModel.GroupBy(gr => CultureInfo.CurrentCulture.Calendar.GetWeekOfYear(
-                gr.Date, CalendarWeekRule.FirstDay, DayOfWeek.Monday))
-                .OrderBy(w => w.Key)
-                .Select(w => new Tuple<int, List<GameViewModel>>(w.Key, w.ToList()))
+            var resultGroupedByWeek = games.GroupBy(GetWeekOfYear)
+                .OrderBy(w => w.Key.Year)
+                .ThenBy(w => w.Key.Week)
+                .Select(w => new Tuple<int, List<GameViewModel>>(w.Key.Week, w.ToList()))
                 .ToList();
 
             var result = new ScheduleViewModel()
```

Comment:
Did you think about making this query more readable?

Reply:
See PR #45